freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/model/FreeIpaUsersState.java

```diff
@@ -0,0 +1,53 @@
+package com.sequenceiq.freeipa.service.freeipa.user.model;
+
+import static java.util.Objects.requireNonNull;
+
+import com.google.common.collect.ImmutableMap;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class FreeIpaUsersState {
+
+    private final UsersState usersState;
+
+    private final ImmutableMap<String, UserMetadata> userMetadataMap;
```

Comment:
Don't we calculate this when generating the users states from both the UMS and the FreeIPA side? If so then this can be pushed into the UsersState.

Reply:
UserMetadata is a tuple of (CRN, workload credentials version). On the UMS side, it seemed very natural to model the credentials version as a property of WorkloadCredential, which we don't have on the IPA side. The CRN, I think, belongs in FmsUser, although it can't go there right now because we don't always know the CRN on the IPA side. (I filed CDPCP-3994 to track refactoring the CRN into FmsUser.) So I'd propose leaving this as is for now, but revisiting it in the context of CDPCP-3994. What do you think?

Reply:
On further thought, it will be a while before the CRN can be moved into FmsUser, and using UserMetadata to store the CRN and workload credential version on both the UMS and IPA sides makes a lot of sense. So I'll go ahead and push this down into UsersState, as you suggested.

Reply:
Done.